### ISSUE TITLE
Cannot Interact with NPC while nicked on Login.

### ISSUE THREAD

**Issue by Chibilena** [MinecraftLibraries/NPCLib#133]
In Paper you have the option to change the player profile such as the nickname or skin via AsyncPlayerPreLoginEvent, but if you use this you can no longer interact with any NPC.

```
		<dependency>
			<groupId>net.jitse</groupId>
			<artifactId>npclib-api</artifactId>
			<version>2.11.1-SNAPSHOT</version>
			<scope>compile</scope>
		</dependency>
```

Plugin-Version: NPCLib 2.11.1-SNAPSHOT
Server-Version: paper-1.16.4-354

```
	@EventHandler //Used Code in Paper as Event...
	public void AsyncPlayerPreLogin(AsyncPlayerPreLoginEvent event) {
 
		UUID uuid = event.getUniqueId();		
		String nickname = "test";
 
		event.setPlayerProfile(Bukkit.createProfile(uuid, nickname));
 
	}
```

```
[09:09:39 ERROR]: Could not pass event PlayerLoginEvent to Citybuild v1.0 //See the Error...
java.lang.NullPointerException: null
        at sun.reflect.UnsafeFieldAccessorImpl.ensureObj(UnsafeFieldAccessorImpl.java:57) ~[?:1.8.0_272]
        at sun.reflect.UnsafeQualifiedObjectFieldAccessorImpl.get(UnsafeQualifiedObjectFieldAccessorImpl.java:38) ~[?:1.8.0_272]
        at java.lang.reflect.Field.get(Field.java:393) ~[?:1.8.0_272]
        at com.comphenix.tinyprotocol.Reflection$1.get(Reflection.java:153) ~[?:?]
        at com.comphenix.tinyprotocol.TinyProtocol.getChannel(TinyProtocol.java:279) ~[?:?]
        at com.comphenix.tinyprotocol.TinyProtocol.access$900(TinyProtocol.java:30) ~[?:?]
        at com.comphenix.tinyprotocol.TinyProtocol$5.onPlayerLogin(TinyProtocol.java:166) ~[?:?]
        at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:37) ~[patched_1.16.4.jar:git-Paper-354]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.4.jar:git-Paper-354]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.4.jar:git-Paper-354]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.4.jar:git-Paper-354]
        at net.minecraft.server.v1_16_R3.PlayerList.attemptLogin(PlayerList.java:667) ~[patched_1.16.4.jar:git-Paper-354]
        at net.minecraft.server.v1_16_R3.LoginListener.c(LoginListener.java:148) ~[patched_1.16.4.jar:git-Paper-354]
        at net.minecraft.server.v1_16_R3.LoginListener.tick(LoginListener.java:66) ~[patched_1.16.4.jar:git-Paper-354]
        at net.minecraft.server.v1_16_R3.NetworkManager.a(NetworkManager.java:385) ~[patched_1.16.4.jar:git-Paper-354]
        at net.minecraft.server.v1_16_R3.ServerConnection.c(ServerConnection.java:145) ~[patched_1.16.4.jar:git-Paper-354]
        at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1381) ~[patched_1.16.4.jar:git-Paper-354]
        at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:377) ~[patched_1.16.4.jar:git-Paper-354]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1209) ~[patched_1.16.4.jar:git-Paper-354]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:997) ~[patched_1.16.4.jar:git-Paper-354]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:178) ~[patched_1.16.4.jar:git-Paper-354]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_272]
```

You can join the server normally, all NPCs will also be shown to you, but you just cannot interact with them. NPCInteractEvent.
I assume, because a new profile is overwritten for the player in the async phase and this is no longer injected by the plugin.

**Comment by Satermelon** [comment]
Hey. I'm assuming you mean nicked as in using the /nick command to nickname yourself. I don't have this bug but I have something very similar. When I login after re-compiling my plugin & restarting the server to implement some changes to my plugin, the NPCs show up, but when I click on them, nothing happens. If I relog, then they do work. I'm not really sure why this happens, I assume also it has something to do with NPCInteract or something. I do have a similar error or maybe the exact same as you where it says could not pass PlayerLoginEvent. Hopefully this issue can be fixed soon because it is quite annoying & I'm not exactly sure why/how this happens, and why it works after you relog. And also another thing, this only happens sometimes & does not happen to all players, just a few including me. Maybe it doesn't work due to the NPC not loading up properly for that 1 player or whoever experiences this? Idk, just a thought.

**Comment by Chibilena** [comment]
This is because the player couldn't inject.

**Comment by Satermelon** [comment]
Wdym couldn't inject? Sorry stupid question

**Comment by JitseB** [comment]
@Satermelon the reload issue you're referring to has to do with improper hiding/showing. The NPCs created before the reload need to be destroyed on plugin disable, else the NPCs will still be visible to players. On enable you should recreate the NPCs and show the appropriately.

@Chibilena the channel lookup used in TinyProtocol is dependent on the player name, I will change this to be UUID based. This will be done shortly and fixed in NPCLib version 2.12-SNAPSHOT.

**Comment by JitseB** [comment]
Closing this issue for now, if this pops up again please don't hesitate to message about this. :)
Happy coding!
